include/nbl/asset/IRayTracingPipeline.h

```diff
@@ -0,0 +1,172 @@
+#ifndef _NBL_ASSET_I_RAY_TRACING_PIPELINE_H_INCLUDED_
+#define _NBL_ASSET_I_RAY_TRACING_PIPELINE_H_INCLUDED_
+
+#include "nbl/asset/IShader.h"
+#include "nbl/asset/IPipeline.h"
+
+#include <span>
+#include <bit>
+#include <type_traits>
+
+namespace nbl::asset
+{
+
+class IRayTracingPipelineBase : public virtual core::IReferenceCounted
+{
+  public:
+    struct SShaderGroupsParams
+    {
+      struct SIndex
+      {
+        constexpr static inline uint32_t Unused = 0xffFFffFFu;
+        uint32_t index = Unused;
+      };
+
+      struct SHitGroup
+      {
+        uint32_t closestHit = SIndex::Unused;
+        uint32_t anyHit = SIndex::Unused;
+        uint32_t intersectionShader = SIndex::Unused;
+      };
+
+      SIndex raygen;
+      std::span<SIndex> misses;
+      std::span<SHitGroup> hits;
+      std::span<SIndex> callables;
+
+      inline uint32_t getShaderGroupCount() const
+      {
+        return 1 + hits.size() + misses.size() + callables.size();
+      }
+
+    };
+    using SGeneralShaderGroup = SShaderGroupsParams::SIndex;
+    using SHitShaderGroup = SShaderGroupsParams::SHitGroup;
+
+    struct SCachedCreationParams final
+    {
+      uint32_t maxRecursionDepth;
```

Comment:
use a bitfield, you only need like 6 bits for the recursion bits, also add another thing
```cpp
// whether we can tell the driver to use a smaller stack size
// https://registry.khronos.org/vulkan/specs/latest/html/vkspec.html#ray-tracing-pipeline-stack
uint32_t dynamicStackSize : 1 = false;
```